src/irmin-git/closeable.ml

```diff
@@ -0,0 +1,47 @@
+(*
+ * Copyright (c) 2013-2017 Thomas Gazagnaire <thomas@gazagnaire.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+module Content_addressable (S : Irmin.CONTENT_ADDRESSABLE_STORE) = struct
+  type 'a t = 'a S.t
+
+  type key = S.key
+
+  type value = S.value
+
+  let closed = ref false
```

Comment:
I don't understand why this is a global value. Should that value be threaded in `a t` instead?

Reply:
it doesn't work, because the type `'a t = G.t` has to match with the type `G.t` after wrapping the store in closeable (line 997 in `irmin-git`, sorry I can't figure out how to add a link to a line in a PR). 
this is way this type is exposed in the `closeable.mli`.

Reply:
But if you create multiple stores and close only one, with the global variable they will all be closed!

Reply:
we can have a look together later, but that would be nice if we can somehow hide what `'a t` is and provide a projection function `'a t -> G.t` to get back an handler to the underlying Git store.

Also `imin_git.ml` should proabably be reviewed/simplified in a future point in time as the current code is just horrible to maintain :-(

Reply:
indeed ! I'll try to add `closed` then in the `'a t` type in the module `Make_ext`

Reply:
It would also nice to have a backend test that closes on different stores do not interfere with each other.

Reply:
> It would also nice to have a backend test that closes on different stores do not interfere with each other.

I added a test like this but for the same type of store, for now. 

Let's add the test you mention in another PR, as for the moment, there are no tests in irmin where stores of different types interfere with one another.

Reply:
Cool; that's what I meant anyway :slightly_smiling_face: 'different stores' was intended to mean 'multiple stores of the same type'